### PR TITLE
Report config to complete json

### PR DESCRIPTION
**Pull request by dannyroberts**
http://manage.dimagi.com/default.asp?175848

### CODE REVIEW

**Comment by dannyroberts** [comment]
@czue @TylerSheffels

File: corehq/apps/reports/views.py [hunk 1]
```diff
@@ -165,7 +165,16 @@ def saved_reports(request, domain, template="reports/reports_home.html"):
...
         raise Http404
 
-    configs = ReportConfig.by_domain_and_owner(domain, user._id)
+    all_configs = ReportConfig.by_domain_and_owner(domain, user._id)
+    good_configs = []
+    for config in all_configs:
+        if config.is_configurable_report:
+            try:
+                config.configurable_report.spec
+            # todo: there's got to be a better thing to raise in this situation
```
Comment (TylerSheffels) [inline corehq/apps/reports/views.py]: I kind of agree w/ this comment.  It's a bit confusing that this is filtering out deleted reports, but if this is the best way you found I'm ok w/ it/

**Comment by TylerSheffels** [comment]
1 comment otherwise good.

File: corehq/apps/reports/models.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 from corehq.apps.domain.middleware import CCHQPRBACMiddleware
 from corehq.apps.export.models import FormQuestionSchema
 from corehq.apps.reports.display import xmlns_to_name
+from corehq.util.couch import DocumentNotFound
```
Comment (dimagimon) [inline corehq/apps/reports/models.py]: F401 'DocumentNotFound' imported but unused

File: corehq/apps/userreports/reports/view.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 from django.conf import settings
 from django.contrib import messages
 from django.core.urlresolvers import reverse
-from django.http import HttpResponse
+from django.http import HttpResponse, Http404
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/view.py]: F401 'Http404' imported but unused

**Comment by dannyroberts** [comment]
@TylerSheffels ok how's that?

**Pull request merged**